### PR TITLE
Fixes Isaac Sim executable on pip installation

### PR DESCRIPTION
**Pull request by Toni-SM**
# Description

This PR fix https://github.com/isaac-sim/IsaacLab/issues/831 by taking into account the Isaac Sim installation using PIP

- Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

**Comment by Mayankm96** [comment]
@Toni-SM Can you also make sure it works on windows?

**Comment by Toni-SM** [comment]
> @Toni-SM Can you also make sure it works on windows?

I updated the PR and tested on Windows

File: isaaclab.bat [hunk 1]
```diff
@@ -81,7 +81,7 @@ if errorlevel 1 (
     set isaacsim_exe=!isaac_path!\isaac-sim.bat
 ) else (
     rem if isaac sim installed from pip
-    set isaacsim_exe=isaacsim
+    set isaacsim_exe=isaacsim omni.isaac.sim
```
Comment (Mayankm96) [inline isaaclab.bat:84]:
```suggestion
    set isaacsim_exe=isaacsim omni.isaac.sim.kit
```

File: isaaclab.sh [hunk 2]
```diff
@@ -90,8 +90,14 @@ extract_isaacsim_exe() {
     local isaacsim_exe=${isaac_path}/isaac-sim.sh
     # check if there is a python path available
     if [ ! -f "${isaacsim_exe}" ]; then
-        echo "[ERROR] No Isaac Sim executable found at path: ${isaacsim_exe}" >&2
-        exit 1
+        # check for installation using Isaac Sim pip
+        if [ $(python -m pip list | grep -c 'isaacsim-rl') -gt 0 ]; then
+            # Isaac Sim - Python packages entry point
+            local isaacsim_exe="isaacsim omni.isaac.sim"
```
Comment (Mayankm96) [inline isaaclab.sh:96]:
```suggestion
            local isaacsim_exe="isaacsim omni.isaac.sim.kit"
```

Comment (Toni-SM) [reply isaaclab.sh:96]: There is no need to add `.kit`
https://docs.omniverse.nvidia.com/isaacsim/latest/installation/install_python.html#launching-isaac-sim-experiences

File: isaaclab.bat (hunk 1 above)
Comment (Toni-SM) [reply isaaclab.bat:84]: There is no need to add `.kit`
https://docs.omniverse.nvidia.com/isaacsim/latest/installation/install_python.html#launching-isaac-sim-experiences

Comment (Toni-SM) [reply isaaclab.bat:84]: Also, in next Isaac Sim release the `omni.isaac.sim.kit` experience file will be the default. So `isaacsim` will launch full Isaac Sim

File: isaaclab.sh (hunk 2 above)
Comment (Toni-SM) [reply isaaclab.sh:96]: Also, in next Isaac Sim release the `omni.isaac.sim.kit` experience file will be the default. So `isaacsim` will launch full Isaac Sim

**Review by kellyguo11 — APPROVED** [review]

**Review by Dhoeller19 — APPROVED** [review]
Thanks @Toni-SM!

**Pull request merged**